sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/UpgradeTest.scala

```diff
@@ -144,13 +177,17 @@ class UpgradeTest(majorLanguageVersion: LanguageMajorVersion)
   }
 
   val pkgName = {
-    assert(pkg1.name == pkg2.name && pkg2.name == pkg3.name && pkg3.name == pkg4.name)
+    assert(
+      pkg0.metadata.map(
+        _.name
+      ) == pkg1.name && pkg1.name == pkg2.name && pkg2.name == pkg3.name && pkg3.name == pkg4.name
+    )
```

Comment:
Why not `pkg0.name`? is `pkg0` a different type to `pkg1/2/3/4`?

Reply:
`Package#name` is defined the package name iff LF >= 1.16. (as it is use in the scala world only in this case).
`Package#metadata` is always define (except for very old LF <= 8).

So here `pkg0.name = None` while `pkg0.metadata.map(_.name) = Some("-upgrade-test-")`

I admit the name of the former method is maybe not very good,  but do you have another idea ?

Reply:
Ah fair enough, yeah you'd probably want something like `upgradesPackageName` and have it separate, but for this case its fine.